app/src/common/shared/com/igalia/wolvic/findinpage/FindInPageInteractor.kt

```diff
@@ -0,0 +1,80 @@
+package com.igalia.wolvic.findinpage
+
+import mozilla.components.browser.state.state.content.FindResultState
+import mozilla.components.feature.findinpage.view.FindInPageView
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.feature.UserInteractionHandler
+import mozilla.components.support.ktx.android.view.hideKeyboard
+import org.mozilla.geckoview.GeckoResult
+import org.mozilla.geckoview.GeckoSession
+import org.mozilla.geckoview.GeckoSession.FINDER_FIND_BACKWARDS
+import org.mozilla.geckoview.SessionFinder
+
+/**
+ * Interactor that implements [FindInPageView.Listener] and notifies about actions the user
+ * performed (e.g. "find next result").
+ */
+class FindInPageInteractor(
+    private val view: FindInPageView,
+    private val onClose: (() -> Unit)? = null,
+) : FindInPageView.Listener, LifecycleAwareFeature, UserInteractionHandler {
+    private var sessionFinder: SessionFinder? = null
+
+    private fun findSession(searchString: String?, flags: Int) {
+        sessionFinder?.find(searchString, flags)?.then { result: GeckoSession.FinderResult? ->
```

Comment:
We cannot use GeckoSession here. This is common code. The rest of the code looks general enough, so I guess you'd have to refactor this somehow. Perhaps adding a FinderResult to WSession for example and then also using WResult instead of GeckoResult

Reply:
Okay, let me check if it's possible.

Reply:
Done!